apis/kueue/v1beta1/workload_types.go

```diff
@@ -0,0 +1,161 @@
+	QueueName string `json:"queueName,omitempty"`
+
+	// If specified, indicates the workload's priority.
+	// "system-node-critical" and "system-cluster-critical" are two special
+	// keywords which indicate the highest priorities with the former being
+	// the highest priority. Any other name must be defined by creating a
+	// PriorityClass object with that name. If not specified, the workload
+	// priority will be default or zero if there is no default.
+	PriorityClassName string `json:"priorityClassName,omitempty"`
+
+	// Priority determines the order of access to the resources managed by the
+	// ClusterQueue where the workload is queued.
+	// The priority value is populated from PriorityClassName.
+	// The higher the value, the higher the priority.
+	// If priorityClassName is specified, priority must not be null.
+	Priority *int32 `json:"priority,omitempty"`
+}
+
+type Admission struct {
+	// clusterQueue is the name of the ClusterQueue that admitted this workload.
+	ClusterQueue ClusterQueueReference `json:"clusterQueue"`
+
+	// podSetFlavors hold the admission results for each of the .spec.podSets entries.
+	// +listType=map
+	// +listMapKey=name
+	PodSetFlavors []PodSetFlavors `json:"podSetFlavors"`
+}
+
+type PodSetFlavors struct {
+	// Name is the name of the podSet. It should match one of the names in .spec.podSets.
+	// +kubebuilder:default=main
+	Name string `json:"name"`
+
+	// Flavors are the flavors assigned to the workload for each resource.
+	Flavors map[corev1.ResourceName]string `json:"flavors,omitempty"`
+}
+
+type PodSet struct {
+	// name is the PodSet name.
+	Name string `json:"name"`
+
+	// template is the Pod template.
+	//
+	// The only allowed fields in template.metadata are labels and annotations.
+	//
+	// If requests are omitted for a container or initContainer,
+	// they default to the limits if they are explicitly specified for the
+	// container or initContainer.
+	//
+	// During admission, the rules in nodeSelector and
+	// nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution that match
+	// the keys in the nodeLabels from the ResourceFlavors considered for this
+	// Workload are used to filter the ResourceFlavors that can be assigned to
+	// this podSet.
+	Template corev1.PodTemplateSpec `json:"template"`
+
+	// count is the number of pods for the spec.
+	// +kubebuilder:validation:Minimum=1
+	Count int32 `json:"count"`
+}
+
+// WorkloadStatus defines the observed state of Workload
+type WorkloadStatus struct {
+	// admission holds the parameters of the admission of the workload by a
+	// ClusterQueue. admission can be set back to null, but its fields cannot be
+	// changed once set.
+	Admission *Admission `json:"admission,omitempty"`
+
+	// conditions hold the latest available observations of the Workload
+	// current state.
+	//
+	// The type of the condition could be:
+	//
+	// - Admitted: the Workload was admitted through a ClusterQueue.
```

Comment:
Extend the list with `PodsReady` for consistency.

Reply:
Added.